Rename Templatizer->Templatize & add bc Templatizer behavior

kevinpschaaf opened the pull request:
Rename `Polymer.Templatizer` library to `Polymer.Templatize` & add pseudo-backwards-compatible `Polymer.Templatizer` behavior for slightly better hybrid support.

For now, users will need to implement both 1.0 & 2.0 callbacks differently (to e.g. deal with path differences due to removal of Collection, different `TemplateInstance` API, etc:

Example hybrid element:
```js
Polymer({
  ...
  behaviors: [ Polymer.Templatizer ],
  // 2.0 callback implementation
  _forwardHostPropV2: function(prop, value) { /* use 2.0 API */ },
  _notifyInstancePropV2: function(inst, prop, value) { /* use 2.0 API */ },
  // 1.0 callback implementation
  _forwardInstancePath function(inst, path, value) { /* use 1.0 API */ },
  _forwardInstanceProp function(inst, prop, value) { /* use 1.0 API */ },
  _forwardParentPath: function(path, value) { /* use 1.0 API */ },
  _forwardParentProp: function(prop, value) { /* use 1.0 API */ },
  ...
});
```

web-padawan commented:
@kevinpschaaf could you check these issues please?
 #3755 #3307

blasten commented on src/templatizer/templatizer.html:
```diff
@@ -298,6 +298,24 @@
+    let Templatizer = {
+      templatize(template) {
+        this._userTemplate = template;
+        this.ctor = Templatize.templatize(template, {
```
I changed `_ctor` to`ctor` since that is how it was defined in v1.

blasten commented:
Having the tests for iron-list 2.0 in hybrid mode passing, I think the changes LGTM. Thanks!

kevinpschaaf commented:
@blasten We treated this PR as an overall review of the 2.0 Templatizer API, and made the following changes:

* Renamed callback `fwdHostPropToInstance` -> `forwardHostProp`
* Renamed callback `fwdInstancePropToHost` -> `notifyInstanceProp`
* Renamed behavior callback `_fwdHostPropToInstance` -> `_forwardHostPropV2`
* Renamed behavior callback `_fwdInstancePropToHost` -> `_notifyInstancePropV2`
* `templatize` now takes 2nd arg for the binding context of the callbacks (normally `this` of the element calling `templatize`)
* removed first argument of `forwardHostProp` callback
* TemplateInstance constructor no longer takes `host` argument
* TemplateInstance API `forwardProperty` & `flushProperties` removed
* TemplateInstance API `forwardHostProp` added, should be called from `forwardHostProp` callback

Refer to e.g. `dom-repeat` implementation for usage changes...
